particula/builder_mixin.py

```diff
@@ -39,7 +40,8 @@ def set_density(
         if density_units == "kg/m^3":
             self.density = density
             return self
-        raise ValueError("Density units must be in kg/m^3")
+        self.density = density * convert_units(density_units, "kg/m^3")
```

Comment:
So, are you thinking don't do this?
Which is fine by me. 

The units input option seems kind a dumb, but maybe should stay for new users.